### PR TITLE
Fixed #12199 -- Added the ability to use "as" with the {% firstof %} template tag.

### PR DESCRIPTION
**Pull request by craigls**

### CODE REVIEW

File: docs/ref/templates/builtins.txt [hunk 1]
```diff
@@ -285,6 +285,11 @@ Or if only some variables should be escaped, you can use::
 
     {% firstof var1 var2|safe var3 "<strong>fallback value</strong>"|safe %}
 
+
```
Comment (timgraham) [inline docs/ref/templates/builtins.txt]: Add `.. versionadded:: 1.9` and a mention in the 1.9 release notes.
Also, only 1 blank line between sections please.

File: tests/template_tests/syntax_tests/test_firstof.py [hunk 2]
```diff
@@ -81,3 +81,24 @@ def test_firstof13(self):
     def test_firstof14(self):
         output = self.engine.render_to_string('firstof14', {'a': '<'})
         self.assertEqual(output, '<')
+
+    @setup({'firstof15': '{% firstof a b c as myvar %}'})
```
Comment (timgraham) [inline tests/template_tests/syntax_tests/test_firstof.py]: It seems to me that 1 test should be sufficient for testing the "as" functionality.

**Comment by timgraham** [comment]
Could you squash commits after making those updates? Thanks!

**Comment by craigls** [comment]
Sure!

**Pull request closed without merging**